src/typechecker.erl

```diff
@@ -1115,33 +1135,68 @@ expect_fun_type_union(Env, [Ty|Tys]) ->
             [TyOut | expect_fun_type_union(Env, Tys)]
     end.
 
-expect_record_type(Record, {type, _, record, [{atom, _, Name}]}, _TEnv) ->
-    if Record == Name ->
-         {ok, constraints:empty()};
-       true ->
-         type_error
+-spec expect_record_type(type(), atom(), #tenv{}) -> Any | ElemTy | ElemTys | TypeError when
+    Any :: any,
+    ElemTy :: {elem_ty, [typed_record_field()], constraints:constraints()},
+    ElemTys :: {elem_tys, [[typed_record_field()]], constraints:constraints()},
```

Comment:
The `elem_ty` name is perhaps not the best choice here. The reason I chose that name was to refer to the type of the elements of lists and tuples. But in this case I think it's more appropriate to call it `fields_ty`.

Reply:
Renamed